algorithms/kernel/k_nearest_neighbors/oneapi/bf_knn_classification_predict_kernel_ucapi_impl.i

```diff
@@ -146,20 +123,15 @@ Status KNNClassificationPredictKernelUCAPI<algorithmFpType>::
 
     const size_t nDataBlocks = nDataRows / dataBlockSize + int (nDataRows % dataBlockSize != 0);
     const size_t nProbeBlocks = nProbeRows / probeBlockSize + int (nProbeRows % probeBlockSize != 0);
-    QuickSelectIndexed::Result selectResult(context, nK, probeBlockSize, distances.type(), categories.type(), &st);
+    SelectIndexed::Result selectResult(context, nK, probeBlockSize, distances.type(), &st);
     DAAL_CHECK_STATUS_VAR(st);
 
-    BlockDescriptor<algorithmFpType> probeRows;
-    DAAL_CHECK_STATUS_VAR(ntData->getBlockOfRows(0, nProbeRows, readOnly, probeRows));
-    auto probes_all = probeRows.getBuffer();
-
-    BlockDescriptor<algorithmFpType> dataRows;
-    DAAL_CHECK_STATUS_VAR(points->getBlockOfRows(0, nDataRows, readOnly, dataRows));
-    auto data_all = dataRows.getBuffer();
-
-    BlockDescriptor<int> labelRows;
-    DAAL_CHECK_STATUS_VAR(labels->getBlockOfRows(0, nDataRows, readOnly, labelRows));
-    auto labels_all = labelRows.getBuffer();
+    SelectIndexed::Params params(nK, TypeIds::id<algorithmFpType>(), dataBlockSize, parameter->engine);
+    SelectIndexedFactory factory;
+    UniquePtr<SelectIndexed, sse2> selector(factory.Create(nK, params, &st));
```

Comment:
It is valid to use sse2, but what the need to use `UniquePtr` here (Why not to use shared pointer)?